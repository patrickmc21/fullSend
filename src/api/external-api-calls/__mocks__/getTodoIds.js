const getToDoIds = jest.fn();

export default getToDoIds;