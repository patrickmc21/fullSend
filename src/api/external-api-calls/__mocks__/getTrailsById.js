import { mockTodos } from '../../../mock-data/mock-data';

const getTrailsById = jest.fn().mockImplementation(() => {
  return Promise.resolve({
    trails: mockTodos
  })
});

export default getTrailsById;