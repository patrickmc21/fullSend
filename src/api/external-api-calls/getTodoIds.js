import key from '../apiKeys/mtb-project-key.js';

const getTodoIds = async (email) => {
  const rootUrl = 'https://www.mtbproject.com/data/get-to-dos';
  const urlOptions = `?email=${email}&key=${key}`;
  const endpoint = `${rootUrl}${urlOptions}`;

  try {
    const response = await fetch(endpoint);
    const ids = await response.json();
    return ids;
  } catch (error) {
    throw error.message;
  };
};

export default getTodoIds;