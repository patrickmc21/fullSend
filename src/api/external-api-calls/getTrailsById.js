import key from '../apiKeys/mtb-project-key.js';

const getTrailsById = async (ids) => {
  console.log(ids);
  const idString = ids.toDos.join(',');
  const rootUrl = 'https://www.mtbproject.com/data/get-trails-by-id';
  const urlOptions = `?ids=${idString}&key=${key}`;
  const endpoint = `${rootUrl}${urlOptions}`;
  
  try {
    const response = await fetch(endpoint);
    const trails = await response.json();
    console.log(trails)
    return trails;
  } catch (error) {
    throw error.message;
  };
};

export default getTrailsById;