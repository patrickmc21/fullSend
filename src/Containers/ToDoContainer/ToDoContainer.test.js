import React from 'react';
import { shallow } from 'enzyme';
import getTrailsById from '../../api/external-api-calls/getTrailsById';
import getTodoIds from '../../api/external-api-calls/getTodoIds';
jest.mock('../../api/external-api-calls/getTrailsById');
jest.mock('../../api/external-api-calls/getTodoIds');

import * as mock from '../../mock-data/mock-data';

import { 
  ToDoContainer, 
  mapStateToProps, 
  mapDispatchToProps } from './ToDoContainer';

describe('ToDoContainer', () => {

  let mockUser;
  let mockTodos;
  let mockAddTodos;
  let wrapper;

  beforeEach(() => {
    mockUser = mock.mockUser;
    mockTodos = mock.mockTodos;
    mockAddTodos = jest.fn();
    wrapper = shallow(
      <ToDoContainer
        user={mockUser}
        todos={mockTodos}
        addTodos={mockAddTodos} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getTodoIds on getToDoRides', () => {
    const expected = mockUser.email;
    wrapper.instance().getToDoRides(mockUser);
    expect(getTodoIds).toHaveBeenCalledWith(expected);
  });

  it('should call getTrailsById on getToDoRides', () => {
    wrapper.instance().getToDoRides(mockUser);
    expect(getTrailsById).toHaveBeenCalled();
  });

  it('should call return trails on getToDoRides', async () => {
    const expected = mockTodos;
    const results = await wrapper.instance().getToDoRides(mockUser);
    expect(results).toEqual(expected);
  });

  it('should call getToDoRides on componentDidMount', () => {
    mockTodos = [];
    wrapper = shallow(
      <ToDoContainer
        user={mockUser}
        todos={mockTodos}
        addTodos={mockAddTodos} />,
      {disableLifecycleMethods: true}
    );
    const expected = mockUser;
    const spy = jest.spyOn(wrapper.instance(), 'getToDoRides');
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalledWith(expected);
  });

  it('should call addTodos on componentDidMount', async () => {
    mockTodos = [];
    wrapper = shallow(
      <ToDoContainer
        user={mockUser}
        todos={mockTodos}
        addTodos={mockAddTodos} />,
      {disableLifecycleMethods: true}
    );
    const expected = mock.mockTodos;
    await wrapper.instance().componentDidMount();
    expect(mockAddTodos).toHaveBeenCalledWith(expected);
  });

});

describe('mapStateToProps', () => {

  let user;
  let todos;
  let mockState;
  let mapped;

  beforeEach(() => {
    user = mock.mockUser;
    todos = mock.mockTodos;
    mockState = {user, todos};
    mapped = mapStateToProps(mockState);
  });

  it('shoud map user to state', () => {
    const expected = user;
    expect(mapped.user).toEqual(expected);
  });

  it('shoud map todos to state', () => {
    const expected = todos;
    expect(mapped.todos).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should map addTodos to props', () => {
    const mockDispatch = jest.fn();
    const todos = mock.mockTodos;
    const expected = {
      type: 'ADD_TODOS',
      todos
    };
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addTodos(todos);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

});