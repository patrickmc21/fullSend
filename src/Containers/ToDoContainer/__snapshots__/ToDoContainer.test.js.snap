// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToDoContainer
    addTodos={[Function]}
    todos={
        Array [
            Object {
              "ascent": 1195,
              "conditionDate": "2018-04-11 12:26:40",
              "conditionDetails": "Mostly Dry",
              "conditionStatus": "All Clear",
              "descent": -2962,
              "difficulty": "black",
              "high": 6803,
              "id": 157369,
              "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "latitude": 38.5819,
              "length": 14.7,
              "location": "Moab, Utah",
              "longitude": -109.4164,
              "low": 4022,
              "name": "Porcupine Rim",
              "starVotes": 259,
              "stars": 4.7,
              "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
              "type": "Featured Ride",
              "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
            },
            Object {
              "ascent": 2035,
              "conditionDate": "2018-04-08 18:35:55",
              "conditionDetails": "Muddy, Snowy, Icy",
              "conditionStatus": "Bad / Closed",
              "descent": -5890,
              "difficulty": "blueBlack",
              "high": 11962,
              "id": 3671983,
              "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
              "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
              "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
              "latitude": 38.4965,
              "length": 36.2,
              "location": "Whitepine, Colorado",
              "longitude": -106.3254,
              "low": 7458,
              "name": "Monarch Crest - IMBA EPIC",
              "starVotes": 179,
              "stars": 4.7,
              "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
              "type": "Featured Ride",
              "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
            },
          ]
    }
    user={
        Object {
            "id": 12,
            "name": "Pat",
            "token": "2y589wyasdfhgfiouqehr8to8",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="todo-card-container"
>
        <ToDoCard
                todo={
                        Object {
                                "ascent": 1195,
                                "conditionDate": "2018-04-11 12:26:40",
                                "conditionDetails": "Mostly Dry",
                                "conditionStatus": "All Clear",
                                "descent": -2962,
                                "difficulty": "black",
                                "high": 6803,
                                "id": 157369,
                                "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                "latitude": 38.5819,
                                "length": 14.7,
                                "location": "Moab, Utah",
                                "longitude": -109.4164,
                                "low": 4022,
                                "name": "Porcupine Rim",
                                "starVotes": 259,
                                "stars": 4.7,
                                "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
                                "type": "Featured Ride",
                                "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
                              }
                }
        />
        <ToDoCard
                todo={
                        Object {
                                "ascent": 2035,
                                "conditionDate": "2018-04-08 18:35:55",
                                "conditionDetails": "Muddy, Snowy, Icy",
                                "conditionStatus": "Bad / Closed",
                                "descent": -5890,
                                "difficulty": "blueBlack",
                                "high": 11962,
                                "id": 3671983,
                                "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
                                "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
                                "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
                                "latitude": 38.4965,
                                "length": 36.2,
                                "location": "Whitepine, Colorado",
                                "longitude": -106.3254,
                                "low": 7458,
                                "name": "Monarch Crest - IMBA EPIC",
                                "starVotes": 179,
                                "stars": 4.7,
                                "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
                                "type": "Featured Ride",
                                "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
                              }
                }
        />
</div>,
      "className": "todo-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <ToDoCard
              todo={
                            Object {
                                          "ascent": 1195,
                                          "conditionDate": "2018-04-11 12:26:40",
                                          "conditionDetails": "Mostly Dry",
                                          "conditionStatus": "All Clear",
                                          "descent": -2962,
                                          "difficulty": "black",
                                          "high": 6803,
                                          "id": 157369,
                                          "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                          "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                          "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                          "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                          "latitude": 38.5819,
                                          "length": 14.7,
                                          "location": "Moab, Utah",
                                          "longitude": -109.4164,
                                          "low": 4022,
                                          "name": "Porcupine Rim",
                                          "starVotes": 259,
                                          "stars": 4.7,
                                          "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
                                          "type": "Featured Ride",
                                          "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
                                        }
              }
/>,
            <ToDoCard
              todo={
                            Object {
                                          "ascent": 2035,
                                          "conditionDate": "2018-04-08 18:35:55",
                                          "conditionDetails": "Muddy, Snowy, Icy",
                                          "conditionStatus": "Bad / Closed",
                                          "descent": -5890,
                                          "difficulty": "blueBlack",
                                          "high": 11962,
                                          "id": 3671983,
                                          "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
                                          "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
                                          "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
                                          "latitude": 38.4965,
                                          "length": 36.2,
                                          "location": "Whitepine, Colorado",
                                          "longitude": -106.3254,
                                          "low": 7458,
                                          "name": "Monarch Crest - IMBA EPIC",
                                          "starVotes": 179,
                                          "stars": 4.7,
                                          "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
                                          "type": "Featured Ride",
                                          "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
                                        }
              }
/>,
          ],
          false,
        ],
        "className": "todo-card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "157369",
          "nodeType": "function",
          "props": Object {
            "todo": Object {
              "ascent": 1195,
              "conditionDate": "2018-04-11 12:26:40",
              "conditionDetails": "Mostly Dry",
              "conditionStatus": "All Clear",
              "descent": -2962,
              "difficulty": "black",
              "high": 6803,
              "id": 157369,
              "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
              "latitude": 38.5819,
              "length": 14.7,
              "location": "Moab, Utah",
              "longitude": -109.4164,
              "low": 4022,
              "name": "Porcupine Rim",
              "starVotes": 259,
              "stars": 4.7,
              "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
              "type": "Featured Ride",
              "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3671983",
          "nodeType": "function",
          "props": Object {
            "todo": Object {
              "ascent": 2035,
              "conditionDate": "2018-04-08 18:35:55",
              "conditionDetails": "Muddy, Snowy, Icy",
              "conditionStatus": "Bad / Closed",
              "descent": -5890,
              "difficulty": "blueBlack",
              "high": 11962,
              "id": 3671983,
              "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
              "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
              "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
              "latitude": 38.4965,
              "length": 36.2,
              "location": "Whitepine, Colorado",
              "longitude": -106.3254,
              "low": 7458,
              "name": "Monarch Crest - IMBA EPIC",
              "starVotes": 179,
              "stars": 4.7,
              "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
              "type": "Featured Ride",
              "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="todo-card-container"
>
          <ToDoCard
                    todo={
                              Object {
                                        "ascent": 1195,
                                        "conditionDate": "2018-04-11 12:26:40",
                                        "conditionDetails": "Mostly Dry",
                                        "conditionStatus": "All Clear",
                                        "descent": -2962,
                                        "difficulty": "black",
                                        "high": 6803,
                                        "id": 157369,
                                        "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                        "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                        "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                        "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                        "latitude": 38.5819,
                                        "length": 14.7,
                                        "location": "Moab, Utah",
                                        "longitude": -109.4164,
                                        "low": 4022,
                                        "name": "Porcupine Rim",
                                        "starVotes": 259,
                                        "stars": 4.7,
                                        "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
                                        "type": "Featured Ride",
                                        "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
                                      }
                    }
          />
          <ToDoCard
                    todo={
                              Object {
                                        "ascent": 2035,
                                        "conditionDate": "2018-04-08 18:35:55",
                                        "conditionDetails": "Muddy, Snowy, Icy",
                                        "conditionStatus": "Bad / Closed",
                                        "descent": -5890,
                                        "difficulty": "blueBlack",
                                        "high": 11962,
                                        "id": 3671983,
                                        "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
                                        "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
                                        "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
                                        "latitude": 38.4965,
                                        "length": 36.2,
                                        "location": "Whitepine, Colorado",
                                        "longitude": -106.3254,
                                        "low": 7458,
                                        "name": "Monarch Crest - IMBA EPIC",
                                        "starVotes": 179,
                                        "stars": 4.7,
                                        "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
                                        "type": "Featured Ride",
                                        "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
                                      }
                    }
          />
</div>,
        "className": "todo-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <ToDoCard
                todo={
                                Object {
                                                "ascent": 1195,
                                                "conditionDate": "2018-04-11 12:26:40",
                                                "conditionDetails": "Mostly Dry",
                                                "conditionStatus": "All Clear",
                                                "descent": -2962,
                                                "difficulty": "black",
                                                "high": 6803,
                                                "id": 157369,
                                                "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                                "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                                "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                                "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                                                "latitude": 38.5819,
                                                "length": 14.7,
                                                "location": "Moab, Utah",
                                                "longitude": -109.4164,
                                                "low": 4022,
                                                "name": "Porcupine Rim",
                                                "starVotes": 259,
                                                "stars": 4.7,
                                                "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
                                                "type": "Featured Ride",
                                                "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
                                              }
                }
/>,
              <ToDoCard
                todo={
                                Object {
                                                "ascent": 2035,
                                                "conditionDate": "2018-04-08 18:35:55",
                                                "conditionDetails": "Muddy, Snowy, Icy",
                                                "conditionStatus": "Bad / Closed",
                                                "descent": -5890,
                                                "difficulty": "blueBlack",
                                                "high": 11962,
                                                "id": 3671983,
                                                "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
                                                "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
                                                "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
                                                "latitude": 38.4965,
                                                "length": 36.2,
                                                "location": "Whitepine, Colorado",
                                                "longitude": -106.3254,
                                                "low": 7458,
                                                "name": "Monarch Crest - IMBA EPIC",
                                                "starVotes": 179,
                                                "stars": 4.7,
                                                "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
                                                "type": "Featured Ride",
                                                "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
                                              }
                }
/>,
            ],
            false,
          ],
          "className": "todo-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "157369",
            "nodeType": "function",
            "props": Object {
              "todo": Object {
                "ascent": 1195,
                "conditionDate": "2018-04-11 12:26:40",
                "conditionDetails": "Mostly Dry",
                "conditionStatus": "All Clear",
                "descent": -2962,
                "difficulty": "black",
                "high": 6803,
                "id": 157369,
                "imgMedium": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                "imgSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                "imgSmallMed": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                "imgSqSmall": "https://img.youtube.com/vi/t5nCdl3OLBo/hqdefault.jpg",
                "latitude": 38.5819,
                "length": 14.7,
                "location": "Moab, Utah",
                "longitude": -109.4164,
                "low": 4022,
                "name": "Porcupine Rim",
                "starVotes": 259,
                "stars": 4.7,
                "summary": "The world famous Porcupine Rim ride; a bone jarring route with a sweet singletrack finish.",
                "type": "Featured Ride",
                "url": "https://www.mtbproject.com/trail/157369/porcupine-rim",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3671983",
            "nodeType": "function",
            "props": Object {
              "todo": Object {
                "ascent": 2035,
                "conditionDate": "2018-04-08 18:35:55",
                "conditionDetails": "Muddy, Snowy, Icy",
                "conditionStatus": "Bad / Closed",
                "descent": -5890,
                "difficulty": "blueBlack",
                "high": 11962,
                "id": 3671983,
                "imgMedium": "https://cdn-files.apstatic.com/mtb/7003758_medium_1436760699.jpg",
                "imgSmall": "https://cdn-files.apstatic.com/mtb/7003758_small_1436760699.jpg",
                "imgSqSmall": "https://cdn-files.apstatic.com/mtb/7003758_sqsmall_1436760699.jpg",
                "latitude": 38.4965,
                "length": 36.2,
                "location": "Whitepine, Colorado",
                "longitude": -106.3254,
                "low": 7458,
                "name": "Monarch Crest - IMBA EPIC",
                "starVotes": 179,
                "stars": 4.7,
                "summary": "One of Colorado's top 5 epic rides - classic high-altitude riding in the Rockies!",
                "type": "Featured Ride",
                "url": "https://www.mtbproject.com/trail/3671983/monarch-crest-imba-epic",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
